src/components/sound-picker/sound-picker.tsx

```diff
@@ -9,15 +9,124 @@ import "./sound-picker.scss";
+      const audioURL = window.URL.createObjectURL(blob);
+      const arrayBuffer = await (await fetch(audioURL)).arrayBuffer();
+      let audioBuffer = await (new AudioContext()).decodeAudioData(arrayBuffer);
+      let channelData = audioBuffer.getChannelData(0);
+
+      // In testing, the API seems to be adding some 'dead time' at the
+      // beginning of the recording. The code below attempts to detect
+      // that initial 'silence', and minimize it in the clip.
+
+      // Value (empirically determined), below which we assume there's only
+      // silence/background noise.
+      const noiseFloorValue = 0.01;
+
+      // Get the index of the first sound sample which has an amplitude above
+      // background noise.
+      const firstSoundIndex = channelData.findIndex( (soundAmplitude) => {
+        const absValue = Math.abs(soundAmplitude);
+        return (absValue > noiseFloorValue);
+      });
+
+      // If initial 'silence' found, then create a new buffer, without the
+      // initial silence, except for the 0.2s of 'silence' just prior to the
+      // sounds of interest.
+      if (firstSoundIndex !== -1) {
+        const extraSamplesOffset =
+          audioBuffer.sampleRate * 0.2; // 200 milliseconds of samples
+        const adjustedSoundIndex =
+          Math.max(0, (firstSoundIndex - extraSamplesOffset));
+        channelData = channelData.slice(firstSoundIndex);
+        audioBuffer = new AudioBuffer({
+          length: audioBuffer.length - adjustedSoundIndex,
+          sampleRate: audioBuffer.sampleRate
+        });
+        audioBuffer.copyToChannel(channelData, 0);
+      }
+
+      // Invoke the callback for recording completion
+      onRecordingCompleted?.(audioBuffer);
+      audioRecordingChunks = [];
+    };
+
+    setMediaRecorder(recorder);
+    setIsReadyToRecord(true);
+    window.alert("Click on the microphone to start/stop recording (up to: 5 seconds).");
+  };
+
+  const doFinishedRecording = () => {
+    clearTimeout(recordingTimerRef.current);
+
+    mediaRecorder?.stop();
+    setIsRecording(false);
+    const soundSelect = soundSelectRef.current;
+    if (soundSelect) { soundSelect.disabled = false; }
+  };
+
+  const onTimedOutRecording = () => {
+    doFinishedRecording();
+  };
+
+  const onMicIconClicked = () => {
+    const maximumRecordingLengthInMilliseconds = 1000 * 5;
+
+    // Ignore this event, if it happens when "(record my own) is NOT selected"
+    if (!isReadyToRecord) { return; }
+
+    if (mediaRecorder?.state === "inactive") {
+      // Use a one-shot timer, to ensure recording does not exceed the maximum length
+      recordingTimerRef.current = setTimeout(onTimedOutRecording, maximumRecordingLengthInMilliseconds);
+
+      audioRecordingChunks = []; // Clear out any old audio data
+      mediaRecorder?.start();
+      setIsRecording(true);
+      const soundSelect = soundSelectRef.current;
+      if (soundSelect) { soundSelect.disabled = true; }
```

Comment:
This kind of approach isn't really recommended in React (except for some rare use cases, usually related to performance). You could just set a state here, something like:
```
setSoundSelectDisabled(false);
```
and use soundSelectDisabled as an attribute in JSX.